uwblueprint/headsupguys
# Add UI for Signup Auth Flow

## Notion ticket link

[Create Account - Email/Password](https://www.notion.so/uwblueprintexecs/Create-Account-Email-Password-1fd0673d1b2547ddac4e80fe50b170d8)



## Implementation description
* Builds off the PR with auth UI components
* Each screen of the signup flow is a `stage` within the signup page, each stage has a title, component, buttonText, and validation that are used to render the current screen. We keep track of the current stage and increment it as necessary to show the correct stage
* Input fields and their validation status are controlled through the Signup component's state. Validation is run `onBlur` for email, name, password fields



## Steps to test
1. `yarn` to install dependencies
2. Go to `/signup`
3. Check that you can go through the flow, moving forwards/backwards and that the bottom dots are updated
4. Check validation: email is required to be an email format, name must be non-empty, password must be at least 8 characters, terms must have the required fields checked off

| Email  | Name | Password | Terms | Success |
| ------------- | ------------- | --- | --- | --- | 
| ![signup-1](https://user-images.githubusercontent.com/51296329/124400164-92505f00-dcee-11eb-9e87-7186473c11ba.PNG) | ![signup-2](https://user-images.githubusercontent.com/51296329/124400170-967c7c80-dcee-11eb-8687-0dfa0c04d7a2.PNG) | ![signup-3](https://user-images.githubusercontent.com/51296329/124400243-0e4aa700-dcef-11eb-9b83-83454ff66aa6.PNG) | ![signup-4](https://user-images.githubusercontent.com/51296329/124400245-0e4aa700-dcef-11eb-9e81-c688b70412df.PNG) | ![signup-5](https://user-images.githubusercontent.com/51296329/124400246-0e4aa700-dcef-11eb-917a-3c184a02e900.PNG) |



## What should reviewers focus on?
* Changes are visible at `/signup`
* The positioning of some of the elements is a little funky (e.g. I'm using padding to position the main content, so when the back button appears/disappears, the content shifts), not sure of the best way to fix it
* Should the text of the checkboxes also be clickable?
* Should we be validating/changing the button disabled status `onChange` or `onBlur`? I initially did it `onBlur` since it was simpler for validation but I realized this might not be intuitive for usage (you need to click out of the input for the button state to update)
* Is `name` a required field? Wasn't sure of validation required on it


## Checklist
